core/src/bundle_stage.rs

```diff
@@ -122,6 +123,53 @@ struct AllExecutionResults {
     pub post_balances: (TransactionBalances, TransactionTokenBalances),
 }
 
+struct BundleReservedSpace {
+    current_tx_block_limit: u64,
+    current_bundle_block_limit: u64,
+    initial_allocated_cost: u64,
+}
+
+impl BundleReservedSpace {
+    fn reset_reserved_cost(&mut self, working_bank: &Arc<Bank>) {
+        self.current_bundle_block_limit = MAX_BLOCK_UNITS;
+        self.current_tx_block_limit = MAX_BLOCK_UNITS - self.initial_allocated_cost;
+
+        working_bank.write_cost_tracker().unwrap().set_limits(
+            MAX_WRITABLE_ACCOUNT_UNITS,
```

Comment:
these might mess with tests or have other unexpected behavior. can we set limit to current limit for values that we don't wanna change?

Reply:
seems like these are internal values, maybe worth exposing as read-only accessors?

Reply:
wdyt about making the inputs Option<u64> and just ignore if None?

i just added the read-only accessors, and it makes the call to set_limits a lot longer

Reply:
https://github.com/jito-foundation/jito-solana/blob/master/core/src/bundle_stage.rs#L1608

do we handle this correctly

Reply:
nvm, that function is called all over the place

i think i'll just make a function that only sets the block cost limit

Reply:
made another update to fix tests

i only reference MAX_BLOCK_UNITS in init now, and preserve previous limit in update